product_price_inquiry/price_inquiry.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2010-2015 Elico Corp (<http://www.elico-corp.com>)
+#    Alex Duan <alex.duan@elico-corp.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from osv import fields, orm, osv
+from tools.translate import _
+import decimal_precision as dp
+
+
+# we use TransientModel becuase this is just temporary search tool.
+class ProductPriceInquiry(orm.TransientModel):
+    _name = "product.price.inquiry"
+    _description = "Product Price Inquiry"
+
+    def action_button_inquiry(self, cr, uid, ids, context=None):
+        '''Do the price inquiry, generate the inquiry results.'''
+        context = context or {}
+
+        # use the first id.
+        if ids and isinstance(ids, (list, tuple)):
+            ids = ids[0]
+        pricelist_obj = self.pool['product.pricelist']
+        pil_obj = self.pool['product.price.inquiry.line']
+
+        inquiry = self.browse(cr, uid, ids, context=context)
+        pricelist_id = inquiry.pricelist_id.id
+        product_id = inquiry.product_id.id
+        qty = inquiry.qty
+
+        qtylist = qty.strip(' ,').split(',')
+        # remove the duplicated qty and null ones
+        qtylist = [i for i in set(qtylist) if i]
+
+        for q in qtylist:
+            if q.isdigit():
+                price = pricelist_obj.price_get(
+                    cr, uid, [pricelist_id], product_id, q)[pricelist_id]
+
+                pil_obj.create(cr, uid, {
+                    'product_id': product_id,
+                    'pricelist_id': pricelist_id,
+                    'qty': q,
+                    'price_unit': price,
+                    'inquiry_id': inquiry.id
+                })
+            else:
+                raise osv.except_osv(
+                    _('Error Input'),
+                    _('Please check if your product qty is correct!\n'
+                        'The right format of input is: '
+                        'quantity1,quantity2,quantity3'))
+        return True
+
+    _columns = {
+        'name': fields.char('Order Reference', size=64),
```

Comment:
@eLBati is right. Looks like you could do with setting _rec_name to 'product_id' or something, and leave out the name field.